### PR TITLE
feat: Add showButton property to InternalDragHandle

### PR DESCRIPTION
**Pull request by johannes-weber**
Use-case: This will be consumed in board components where we need to render the arrow icons under some conditions on the initial render.

### Description

Related links, issue #, if available: n/a

### How has this been tested?

<details>
   <summary>Review checklist</summary>

_The following items are to be evaluated by the author(s) and the reviewer(s)._

#### Correctness

- _Changes include appropriate documentation updates._
- _Changes are backward-compatible if not indicated, see [`CONTRIBUTING.md`](https://github.com/cloudscape-design/components/blob/main/CONTRIBUTING.md#public-apis)._
- _Changes do not include unsupported browser features, see [`CONTRIBUTING.md`](https://github.com/cloudscape-design/components/blob/main/CONTRIBUTING.md#browsers-support)._
- _Changes were manually tested for accessibility, see [accessibility guidelines](https://cloudscape.design/foundation/core-principles/accessibility/)._

#### Security

- _If the code handles URLs: all URLs are validated through [the `checkSafeUrl` function](https://github.com/cloudscape-design/components/blob/main/src/internal/utils/check-safe-url.ts)._

#### Testing

- _Changes are covered with new/existing unit tests?_
- _Changes are covered with new/existing integration tests?_
</details>

---

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: src/internal/components/drag-handle-wrapper/__tests__/drag-handle-wrapper.test.tsx [hunk 1]
```diff
@@ -29,11 +29,28 @@ function getDirectionButton(direction: Direction) {
...
-  expect(getDirectionButton(direction)?.parentElement).toHaveClass(motionExitingClass);
+// Direction buttons get hidden via transition which doesn't end in JSDOM, so we use
+// the "exiting", "exited" or "hidden" classname instead to verify it's hidden.
+const DIRECTION_BUTTON_HIDDEN_CLASSES = [
+  styles['direction-button-wrapper-motion-exiting'],
+  styles['direction-button-wrapper-motion-exited'],
+  styles['direction-button-wrapper-hidden'],
+];
+
+function expectDirectionButtonToBeHidden(direction: Direction) {
```
Comment (johannes-weber) [inline src/internal/components/drag-handle-wrapper/__tests__/drag-handle-wrapper.test.tsx:40]: In case a direction button is active but not visible, it is rendered in the DOM to be ready for the transition - but set to invisible.

I also included the `*-motion-exiting` and `*-motion-exited` classes in that utility for more general test usage.

File: src/internal/components/drag-handle-wrapper/portal-overlay.tsx [hunk 2]
```diff
@@ -16,14 +16,14 @@ export default function PortalOverlay({
   isDisabled,
   children,
 }: {
-  track: HTMLElement | null;
+  track: React.RefObject<HTMLElement | null>;
```
Comment (johannes-weber) [inline src/internal/components/drag-handle-wrapper/portal-overlay.tsx:19]: We need to pass the `RefObject` instead of the `HTMLElement`.

Why? - During testing the `showButtons=true` flag, I discovered that `track` was all the time `null` on when rendering, no arrow buttons have been shown.
This was because `dragHandleRef` on the parent component was null on the initial render (passed null as `track` property). Once DOM element get assigned to the `dragHandleRef` in the parent component's, it won't trigger a re-render of `PortalOverlay` and it stayed null.

File: src/internal/components/drag-handle-wrapper/portal-overlay.tsx [hunk 3]
```diff
@@ -36,19 +36,23 @@ export default function PortalOverlay({
       // It could be that the portal hasn't been attached to the DOM yet - ensure the ref exists and is attached DOM tree.
       if (ref.current && document.body.contains(ref.current)) {
         const isRtl = getIsRtl(ref.current);
-        const { insetInlineStart, insetBlockStart, inlineSize, blockSize } = getLogicalBoundingClientRect(track);
+        const { insetInlineStart, insetBlockStart, inlineSize, blockSize } = getLogicalBoundingClientRect(
+          track.current!
```
Comment (johannes-weber) [inline src/internal/components/drag-handle-wrapper/portal-overlay.tsx:40]: `!` becaues we know that `track.current` won't be null because we checked it in line 26.
Not sure why TypeScript doesn't consider that check above.

Comment (avinashbot) [reply src/internal/components/drag-handle-wrapper/portal-overlay.tsx:40]: It's because it's in a function and the current could be unset at any point. Perhaps you could add a check for `track.current` inside the if condition on line 37.

File: src/internal/components/drag-handle-wrapper/interfaces.ts [hunk 4]
```diff
@@ -11,4 +11,5 @@ export interface DragHandleWrapperProps {
   tooltipText?: string;
   children: React.ReactNode;
   triggerMode?: TriggerMode;
+  showButtons?: boolean;
```
Comment (avinashbot) [inline src/internal/components/drag-handle-wrapper/interfaces.ts:14]: So this describes the starting state rather than the active value? Perhaps `initialShowButtons` or `initialOpen` would be better?

Comment (johannes-weber) [reply src/internal/components/drag-handle-wrapper/interfaces.ts:14]: That's right. I like the suggestion and think the "initial" in it makes it clear! Updating that.

File: src/internal/components/drag-handle-wrapper/portal-overlay.tsx [hunk 5]
```diff
@@ -36,19 +36,23 @@ export default function PortalOverlay({
...
+        );
         // For simplicity, we just make all our calculations independent of
         // the browser's scrolling edge. When it comes to applying the changes,
         // translate is independent of writing direction, so we need to invert
         // the X coordinate ourselves just before applying the values.
         const newX = (insetInlineStart + getScrollInlineStart(document.documentElement)) * (isRtl ? -1 : 1);
         const newY = insetBlockStart + document.documentElement.scrollTop;
-        if (lastX !== newX || lastY !== newY) {
+        // Check if the current ref has style updates already applied (in case the ref changes)
```
[the remark this answers is not in the document]
Comment (johannes-weber) [reply src/internal/components/drag-handle-wrapper/portal-overlay.tsx]: I cross checked the issue which I had before - I discovered an edge case when I used the `PortalOverlay` in a test component (where I switched the `track` for testing in runtime). Since this is not a use-case, I reverted the changes.

[the remark this answers is not in the document]
Comment (avinashbot) [reply src/internal/components/drag-handle-wrapper/portal-overlay.tsx]: But the current code should still be fine if track changes, just not if ref changes, right?

File: src/internal/components/drag-handle-wrapper/index.tsx [hunk 6]
```diff
@@ -156,7 +157,7 @@ export default function DragHandleWrapper({
       setShowButtons(false);
     } else if (triggerMode === 'keyboard-activate' && (event.key === 'Enter' || event.key === ' ')) {
       // toggle buttons when Enter or space is pressed in 'keyboard-activate' triggerMode
-      setShowButtons(prevShowButtons => !prevShowButtons);
+      setShowButtons(prevshowButtons => !prevshowButtons);
```
Comment (avinashbot) [inline src/internal/components/drag-handle-wrapper/index.tsx:160]: Super minor: the previous one seems more correct...

**Review by avinashbot — APPROVED** [review]

File: src/internal/components/drag-handle-wrapper/index.tsx (hunk 6 above)
Comment (johannes-weber) [reply src/internal/components/drag-handle-wrapper/index.tsx:160]: I'll update that in a separate clean up PR. I'll go ahead and merge this one first.
